### PR TITLE
CBS-843: Add Snapshot quota hard limit code change and unit test in lunrdriver

### PR DESCRIPTION
**Pull request by mona-nanda**

### CODE REVIEW

**Comment by sudhakar-datta** [comment]
Mona, you need remove two lines, 185 and 186 in lunrdriver/api.py. We do not need another get_all call there.

**Review by sudhakar-datta — APPROVED** [review]
Looks Good overall, but require some minor/optional changes

File: lunrdriver/lunr/api.py [hunk 1]
```diff
@@ -46,6 +46,12 @@ class CloneConflict(exception.Invalid):
     code = 409
 
 
+class SnapshotQuotaExceedConflict(exception.Invalid):
```
Comment (sudhakar-datta) [inline lunrdriver/lunr/api.py:49]: I prefer to be just SnapshotQuotaExceed. Conflict denotes something already in progress.

code 429 may be better, or 400 with message that it exceeded the total limit. 413 is mostly for payload related issues. Might need to check with Dan or Tyler as well

**Review by sudhakar-datta — CHANGES_REQUESTED** [review]

File: lunrdriver/lunr/api.py (hunk 1 above)
Comment (mona-nanda) [reply lunrdriver/lunr/api.py]: added status code 429 for the conflict SnapshotExceeded

**Review by sudhakar-datta — APPROVED** [review]
LGTM

**Review by jtwatson46 — APPROVED** [review]
LGTM

**Pull request merged**